### PR TITLE
Fix: handle datetime-local not supported by browser

### PR DESCRIPTION
**Pull request by VirginiaBalseiro**
This PR introduces changes to the `Value` component to better handle the case where `datetime-local` is not supported by the browser by rendering two inputs with type `date` and `time`.

- handle case where `datetime-local` is not supported by browser

- simplify value state to use strings to keep inputs controlled

- add tests

This PR fixes #SDK-2041

- [x] I've added a unit test to test for potential regressions of this bug.
- [x] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

# New feature description

# Checklist

- [x] All acceptance criteria are met.
- [ ] Relevant documentation, if any, has been written/updated.
- [x] The changelog has been updated, if applicable.
- [ ] New functions/types have been exported in `index.ts`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

This PR bumps the version to <version number>.

# Checklist

- [ ] I used `npm version <major|minor|patch>` to update `package.json`, inspecting the changelog to determine if the release was major, minor or patch.
- [ ] The CHANGELOG has been updated to show version and release date - https://keepachangelog.com/en/1.0.0/.
- [ ] `@since X.Y.Z` annotations have been added to new APIs.
- [ ] The **only** commits in this PR are:
  - the CHANGELOG update.
  - the version update.
  - `@since` annotations.
- [ ] I will make sure **not** to squash these commits, but **rebase** instead.
- [ ] Once this PR is merged, I will push the tag created by `npm version ...` (e.g. `git push origin vX.Y.Z`).

### CODE REVIEW

File: src/components/value/index.test.tsx [hunk 1]
```diff
@@ -51,6 +60,11 @@ const inputOptions = {
   type: "url",
 };
 
+const mockDatasetWithBdayResourceInfo = SolidFns.setThing(
+  SolidFns.createSolidDataset() as any,
```
Comment (Vinnl) [inline src/components/value/index.test.tsx:64]: Shouldn't be necessary I think?

```suggestion
  SolidFns.createSolidDataset(),
```

File: src/components/value/index.test.tsx [hunk 2]
```diff
@@ -194,6 +235,34 @@ describe("<Value /> component functional testing", () => {
     }
   );
 
+  it("when dataType is datetime and datetime-local is unssuported, it calls the setDatetime with the correct value", () => {
```
Comment (Vinnl) [inline src/components/value/index.test.tsx:238]:
```suggestion
  it("when dataType is datetime and datetime-local is unsuported, it calls the setDatetime with the correct value", () => {
```

File: src/components/value/index.test.tsx [hunk 3]
```diff
@@ -194,6 +235,34 @@ describe("<Value /> component functional testing", () => {
...
+        thing={mockThing}
+        property={mockBdayPredicate}
+        edit
+        autosave
+        inputProps={{ title: "test title" }}
+      />
+    );
+    const input = getByTitle("test title");
+    input.focus();
+    fireEvent.change(input, { target: { value: "2020-12-30T12:30" } });
```
Comment (Vinnl) [inline src/components/value/index.test.tsx:261]: This doesn't seem right - as I read this, this sets the value of the `date` input to a complete `datetime` string and ignores the `time` input? The test also doesn't validate that `mockSetter` actually was called with the correct value (which I'm assuming would indicate that the time part is being ignored?).

**Review by Vinnl — COMMENTED** [review]
There's a lot of things in this component that do different things depending on what the type of the value is, which means you have to keep track of a lot of things in the back of your mind while reading this code - which I had trouble with :sweat_smile: A suggested refactoring would be to just make this a wrapper of individual components for the different types.

However, since that's unrelated to the problem this PR is trying to solve, I'd be OK with going ahead with this without that too. I've left some small comments/suggestions in line, of which I think only the capitalisation of the hook really and validation pattern is required, and the test needs some work (assuming I understood it correctly).

**Comment by VirginiaBalseiro** [comment]
> There's a lot of things in this component that do different things depending on what the type of the value is, which means you have to keep track of a lot of things in the back of your mind while reading this code - which I had trouble with 😅 A suggested refactoring would be to just make this a wrapper of individual components for the different types.
> 
> However, since that's unrelated to the problem this PR is trying to solve, I'd be OK with going ahead with this without that too. I've left some small comments/suggestions in line, of which I think only the capitalisation of the hook really and validation pattern is required, and the test needs some work (assuming I understood it correctly).

Agreed. I will make this refactor in a separate PR.

File: src/components/value/index.tsx [hunk 4]
```diff
@@ -263,6 +307,31 @@ export function Value({
...
+            {...inputProps}
+            onChange={(e) => {
+              setDate(e.target.value);
+            }}
+            onBlur={(e) => autosave && saveHandler(e)}
+            value={date}
+            pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}"
+            placeholder="yyyy-mm-dd"
+          />
+          <input
```
Comment (Vinnl) [inline src/components/value/index.tsx:326]: Probably useful to add a `pattern` and a `placeholder` to the time input too.

**Review by Vinnl — APPROVED** [review]
One inconsequential suggestion; good to go as far as I'm concerned!

File: src/components/value/index.tsx (hunk 4 above)
Comment (VirginiaBalseiro) [reply src/components/value/index.tsx:326]: Added a `pattern`, but `placeholder` does not work with time input.

Comment (Vinnl) [reply src/components/value/index.tsx:326]: I suppose it's mostly for browsers that do not support it, and thus fall back to `text`? Either way I don't think those are very common, so a minor issue really.

**Pull request merged**